napi-inl.h

```diff
@@ -1373,17 +1373,17 @@ inline void Buffer<T>::EnsureInfo() const {
 inline Error Error::New(napi_env env) {
   napi_status status;
   napi_value error = nullptr;
-  if (Napi::Env(env).IsExceptionPending()) {
-    status = napi_get_and_clear_last_exception(env, &error);
-    assert(status == napi_ok);
-  }
-  else {
-    // No JS exception is pending, so check for NAPI error info.
-    const napi_extended_error_info* info;
-    status = napi_get_last_error_info(env, &info);
-    assert(status == napi_ok);
 
-    if (status == napi_ok) {
+  const napi_extended_error_info* info;
+  status = napi_get_last_error_info(env, &info);
+  assert(status == napi_ok);
+
+  if (status == napi_ok) {
```

Comment:
If we encounter another error while handling an error, there's not much we can do. Probably it should be a fatal error, except we haven't exposed `node::FatalError()` via N-API [yet](https://github.com/nodejs/abi-stable-node/issues/232). The assert will at least indicate when something went wrong in a debug build, but in a release build the conditional allows it to continue without crashing.